tests/regression/87-once/04-thread.c

```diff
@@ -0,0 +1,58 @@
+// PARAM: --set ana.activated[+] pthreadOnce
+#include <pthread.h>
+#include <stdio.h>
+
+int g;
+int h;
+int i;
+pthread_once_t once = PTHREAD_ONCE_INIT;
+pthread_once_t i_once = PTHREAD_ONCE_INIT;
+pthread_mutex_t mut;
+
+void *t_other(void* arg) {
+  g = 17; //RACE
+
+  pthread_mutex_lock(&mut);
+  i = 7; //NORACE
+  pthread_mutex_unlock(&mut);
```

Comment:
What is the mutex for? Against self-races?
If `fun` is run once, then `t_other` should be unique and not self-race.

Or is it that `fun` being run once from different threads causes two different unique thread IDs which would then race here? That seems like imprecision on our part that's worth a TODO.